### PR TITLE
Refactor Pretrained Model Aggregator

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Refactored the Code to be consistent with the Basic FL App
Introduced the notions of 
- launch
- running
- done 
folders similar to the Basic FL App

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -7,6 +8,15 @@
 import re
 import json
 
+APP_NAME = "pretrained_model_aggregator"
```
Comment (khoaguin) [inline main.py]: should this be `API_NAME` to make it consistent with SyftBox API?

File: main.py [hunk 2]
```diff
@@ -33,8 +43,28 @@ def get_model_file(path: Path):
     return model_files[0] if model_files else None
```
Comment (khoaguin) [inline main.py]: we only get the first model if the user has multiple pretrained models?

Comment (rasswanth-s) [reply main.py]: we could discuss this in our sync

File: main.py [hunk 3]
```diff
@@ -65,83 +97,120 @@ def aggregate_model(
...
-            all_users.append(entry)
-
-    participants = list(set(participants) & set(all_users))
-    missing_participants = list(set(participants) - set(all_users))
-    print(f"Pretrained model aggregator participants: {participants}")
-    return participants
+        return None, None
+
+
+def advance_pretrained_aggregator(client: Client) -> None:
```
Comment (shubham3121) [inline main.py]: Can we rename this to something more intuitive `perform_model_weight_aggregation` ?

File: main.py [hunk 4]
```diff
@@ -65,83 +97,120 @@ def aggregate_model(
...
+    global_model, missing_peers = aggregate_model(
+        client,
+        participants,
+        model_output_path
+    )
 
+    if not global_model:
+        print("No models found to aggregate in participants:", participants)
+        return
+
```
Comment (shubham3121) [inline main.py]: can we shift evaluation to a different function: `evaluate_global_model`

**Pull request merged**